components/hidden_data_box/commons/hidden_data_box.lua

```diff
@@ -48,17 +52,11 @@ function HiddenDataBox.run(args)
 		if not queryResult[1] and Namespace.isMain() then
 			table.insert(warnings, String.interpolate(INVALID_PARENT, {parent = parent}))
 		elseif args.participantGrabber then
-			local participants = HiddenDataBox._fetchParticipants(parent)
-
-			Table.iter.forEachPair(participants, function (participant, players)
-				-- TODO: An improvement would be called TeamCard module for this
-				-- Would need a rework for the function that does it however
-				local participantResolved = mw.ext.TeamLiquidIntegration.resolve_redirect(participant)
+			local date = HiddenDataBox.cleanDate(args.date, args.sdate) or queryResult.startdate or
+				Variables.varDefault('tournament_startdate') or HiddenDataBox.cleanDate(args.edate) or
+				queryResult.enddate or Variables.varDefault('tournament_enddate')
 
-				Table.iter.forEachPair(players, function(key, value)
-					HiddenDataBox.setWikiVariableForParticipantKey(participant, participantResolved, key, value)
-				end)
-			end)
+			Array.forEach(HiddenDataBox._fetchPlacements(parent), HiddenDataBox._setWikiVariablesFromPlacement, date)
```

Comment:
Array.forEach only accpects 2 params

Use FnUtil.curry if you want to lock another param